### PR TITLE
Tax rates filter by dates

### PR DESCRIPTION
**Pull request by everwhatever**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12  |
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no |
| Deprecations?   | no |
| Related tickets | [jira ](https://sylius.atlassian.net/browse/SYL-2278)  |
| License         | MIT |
| Blocked by [this pr](https://github.com/Sylius/Sylius/pull/14338)|

### CODE REVIEW

File: src/Sylius/Component/Taxation/composer.json [hunk 1]
```diff
@@ -31,6 +31,8 @@
         "php": "^8.0",
         "sylius/calendar": "^0.3 || ^0.4",
         "sylius/registry": "^1.5",
+        "sylius/calendar": "^0.3",
```
Comment (GSadee) [inline src/Sylius/Component/Taxation/composer.json:34]: It has been already added two lines above

File: src/Sylius/Bundle/TaxationBundle/composer.json [hunk 2]
```diff
@@ -33,7 +33,8 @@
         "sylius/registry": "^1.5",
         "sylius/resource-bundle": "^1.9 || ^1.10@alpha",
         "sylius/taxation": "^1.6",
-        "symfony/framework-bundle": "^5.4 || ^6.0"
+        "symfony/framework-bundle": "^5.4 || ^6.0",
+        "sylius/calendar": "^0.3"
```
Comment (GSadee) [inline src/Sylius/Bundle/TaxationBundle/composer.json:37]: It has been already added above

File: src/Sylius/Bundle/AdminBundle/Resources/config/grids/tax_rate.yml [hunk 3]
```diff
@@ -29,11 +29,21 @@ sylius_grid:
...
+                    type: date
+                    label: sylius.ui.end_date
+                    options:
+                        inclusive_to: true
                 search:
                     type: string
                     label: sylius.ui.search
                     options:
-                        fields: [code, name]
+                        fields: [ code, name ]
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/config/grids/tax_rate.yml:46]: Revert this change

**Comment by Zales0123** [comment]
Merged in https://github.com/Sylius/Sylius/pull/14427 🖖
